geoh5py/shared/concatenation/drillholes_group_table.py

```diff
@@ -205,6 +206,18 @@ def add_values_to_property_group(
                 f"({self.parent.data[self.association[0]].shape})."
             )
 
+        if isinstance(value_map, dict):
+            new_type = DataType(
+                self.parent.workspace, primitive_type="REFERENCED", value_map=value_map
+            )
+        else:
+            new_type = DataType.find_or_create(
+                self.parent.workspace,
+                **DataType.validate_data_type(
+                    self.parent.workspace, {"name": name, "values": values}
+                ),
+            )
```

Comment:
That's great - one step closer to have those methods consolidated. Feels like eventually the two may be combined, but would require a deeper look at where they are used.

Reply:
I combined the two.
Tests were passing on geoh5py, but when I tested with GA it was not able to recognize value_map.
I went back to dic attribute assignations.